systemverilog/SystemVerilog.g4

```diff
@@ -0,0 +1,3558 @@
+	;
+
+package_declaration
+	: attribute_instance* 'package' lifetime? package_identifier ';' timeunits_declaration? (attribute_instance* package_item)* 'endpackage' (':' package_identifier)?
+	;
+
+timeunits_declaration
+	: 'timeunit' TIME_LITERAL (('/' TIME_LITERAL)? | (';' 'timeprecision' TIME_LITERAL)) ';'
+	| 'timeprecision' TIME_LITERAL ';' ('timeunit' TIME_LITERAL ';')
+	;
+
+// A.1.3 Module parameters and ports
+
+parameter_port_list
+	: '#' '(' ((list_of_param_assignments | parameter_port_declaration) (',' parameter_port_declaration)*)? ')'
+	;
+
+parameter_port_declaration
+	: parameter_declaration
+	| local_parameter_declaration
+	| data_type list_of_param_assignments
+	| 'type' list_of_type_assignments
+	;
+
+list_of_ports
+	: '(' port (',' port)* ')'
+	;
+
+list_of_port_declarations
+	: '(' (attribute_instance* ansi_port_declaration (',' attribute_instance* ansi_port_declaration)*)? ')'
+	;
+
+port_declaration
+	: attribute_instance* (inout_declaration | input_declaration | output_declaration | ref_declaration | interface_port_declaration)
+	;
+
+port
+	: port_expression?
+	| '.' port_identifier '(' port_expression? ')'
+	;
+
+port_expression
+	: port_reference
+	| '{' port_reference (',' port_reference)* '}'
+	;
+
+port_reference
+	: port_identifier constant_select
+	;
+
+port_direction
+	: 'input'
+	| 'output'
+	| 'inout'
+	| 'ref'
+	;
+
+net_port_header
+	: port_direction? net_port_type
+	;
+
+variable_port_header
+	: port_direction? variable_port_type
+	;
+
+interface_port_header
+	: interface_identifier ('.' modport_identifier)?
+	| 'interface' ('.' modport_identifier)?
+	;
+
+ansi_port_declaration
+	: (net_port_header | interface_port_header)? port_identifier unpacked_dimension* ('=' constant_expression)?
+	| variable_port_header? port_identifier variable_dimension* ('=' constant_expression)?
+	| port_direction? '.' port_identifier '(' expression? ')'
+	;
+
+// A.1.4 Module items
+
+elaboration_system_task
+	: '$fatal' ('(' /*finish_number*/ (/*','*/ list_of_arguments)? ')')? ';'
+	| ('$error' | '$warning' | '$info') ('(' list_of_arguments? ')')? ';'
+	;
+
+/*finish_number
+	: '0'
+	| '1'
+	| '2'
+	;
+*/
+module_common_item
+	: module_or_generate_item_declaration
+	| interface_instantiation
+	| program_instantiation
+	| assertion_item
+	| bind_directive
+	| continuous_assign
+	| net_alias
+	| initial_construct
+	| final_construct
+	| always_construct
+	| loop_generate_construct
+	| conditional_generate_construct
+	| elaboration_system_task
+	;
+
+module_item
+	: port_declaration ';'
+	| non_port_module_item
+	;
+
+module_or_generate_item
+	: attribute_instance* (parameter_override | gate_instantiation /*| udp_instantiation*/ | module_instantiation | module_common_item)
+	;
+
+module_or_generate_item_declaration
+	: package_or_generate_item_declaration
+	| genvar_declaration
+	| clocking_declaration
+	| 'default' (('clocking' clocking_identifier) | ('disable' 'iff' expression_or_dist)) ';'
```

Comment:
Also reduntant parentheses:

```suggestion
	| 'default' ('clocking' clocking_identifier | 'disable' 'iff' expression_or_dist) ';'
```

`|` operator has a priority lesser than concatenation priority. Thus there is no need to use parentheses. Here and below in the grammar (I marked seems like all places).